docs/blog/2023-03-31-zephyr-3-2.md

```diff
@@ -0,0 +1,303 @@
+
+#### Out of Tree Boards/Shields
+
+All of the in-tree boards and shields have been upgrade, but if you maintain/use and out-of-tree shield that uses the converted boards and overrides pins for various buses, you may need to switch to `pinctrl` to match ZMK's new approach.
+
+The approach is the following when updating a _board_:
+
+1. Add an entry `CONFIG_PINCTRL=y` to the `<board>_defconfig` file in the board directory.
+1. Add a new file with the naming convention `<board>-pinctrl.dtsi` to your board directory.
+1. In the new file, add your `pinctrl` entries that set up different pin control configurations for whatever peripherals/busses are needed. Here's the nice!nano file as an example:
+
+   ```
+   /*
+   * Copyright (c) 2022 The ZMK Contributors
+   * SPDX-License-Identifier: MIT
+   */
+
+   &pinctrl {
+           uart0_default: uart0_default {
+                   group1 {
+                           psels = <NRF_PSEL(UART_RX, 0, 8)>;
+                           bias-pull-up;
+                   };
+                   group2 {
+                           psels = <NRF_PSEL(UART_TX, 0, 6)>;
+                   };
+           };
+
+           uart0_sleep: uart0_sleep {
+                   group1 {
+                           psels = <NRF_PSEL(UART_RX, 0, 8)>,
+                                   <NRF_PSEL(UART_TX, 0, 6)>;
+                           low-power-enable;
+                   };
+           };
+
+           i2c0_default: i2c0_default {
+                   group1 {
+                           psels = <NRF_PSEL(TWIM_SDA, 0, 17)>,
+                                   <NRF_PSEL(TWIM_SCL, 0, 20)>;
+                   };
+           };
+
+           i2c0_sleep: i2c0_sleep {
+                   group1 {
+                           psels = <NRF_PSEL(TWIM_SDA, 0, 17)>,
+                                   <NRF_PSEL(TWIM_SCL, 0, 20)>;
+                           low-power-enable;
+                   };
+           };
+   };
+   ```
+
+1. From the main `<board>.dts` file, add an `#include "<board>-pinctrl.dtsi"` to have the C-preprocessor combine the files.
+1. Update the various peripheral nodes to use the new `pinctrl` configurations. For example, the following old configuration:
+
+   ```
+   &i2c0 {
+       compatible = "nordic,nrf-twi";
+       sda-pin = <15>;
+       scl-pin = <17>;
+   };
+   ```
+
+   would be changed to:
+
+   ```
+   &i2c0 {
+       compatible = "nordic,nrf-twi";
+       pinctrl-0 = <&i2c0_default>;
+       pinctrl-1 = <&i2c0_sleep>;
+       pinctrl-names = "default", "sleep";
+   };
+   ```
+
+Because `pinctrl` configuration is very dependant on the specific target SoC, you will rarely need to consider it for a shield overlay that leverages a pro micro or XIAO abstraction. As noted, you're more likely to need to fix up pinctrl settings is using a board specific shield overlay, e.g. `<shield>/boards/<board>.overlay` to set things up.
```

Comment:
```suggestion
Because `pinctrl` configuration is very dependent on the specific target SoC, you will rarely need to consider it for a shield overlay that leverages a pro micro or XIAO abstraction. As noted, you're more likely to need to fix up pinctrl settings is using a board specific shield overlay, e.g. `<shield>/boards/<board>.overlay` to set things up.
```